pyccl/pyutils.py

```diff
@@ -1,23 +1,50 @@
 """Utility functions to analyze status and error messages passed from CCL, as
-well as wrappers to automatically vectorize functions."""
-from . import ccllib as lib
-from ._types import error_types
-from .parameters import spline_params
-from .errors import CCLError
+well as wrappers to automatically vectorize functions.
+"""
+__all__ = (
+    "CLevelErrors", "ExtrapolationMethods", "IntegrationMethods", "check",
+    "debug_mode", "get_pk_spline_lk", "get_pk_spline_a", "resample_array")
+
+from typing import Iterable
+
 import numpy as np
-from collections.abc import Iterable
 
-NoneArr = np.array([])
+from . import CCLError, lib, spline_params
 
-integ_types = {'qag_quad': lib.integration_qag_quad,
-               'spline': lib.integration_spline}
 
-extrap_types = {'none': lib.f1d_extrap_0,
-                'constant': lib.f1d_extrap_const,
-                'linx_liny': lib.f1d_extrap_linx_liny,
-                'linx_logy': lib.f1d_extrap_linx_logy,
-                'logx_liny': lib.f1d_extrap_logx_liny,
-                'logx_logy': lib.f1d_extrap_logx_logy}
+NoneArr = np.array([])
+
+IntegrationMethods = {
+    'qag_quad': lib.integration_qag_quad,
+    'spline': lib.integration_spline}
+
+ExtrapolationMethods = {
+    'none': lib.f1d_extrap_0,
+    'constant': lib.f1d_extrap_const,
+    'linx_liny': lib.f1d_extrap_linx_liny,
+    'linx_logy': lib.f1d_extrap_linx_logy,
+    'logx_liny': lib.f1d_extrap_logx_liny,
+    'logx_logy': lib.f1d_extrap_logx_logy}
+
+# This is defined here instead of in `errors.py` because SWIG needs `CCLError`
+# from `.errors`, resulting in a cyclic import.
+CLevelErrors = {
```

Comment:
I'm guessing it breaks API

Reply:
No, it was in `_types`, which was a private module. Private stuff can change any time, they aren't supposed to be used.